src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -75,105 +75,150 @@ public class Session {
-                _sessionLogger.LogTrace(Resources.Trace_EnvironmentVariable, "HOMEDRIVE", psi.EnvironmentVariables["HOMEDRIVE"]);
+                eb["R_HOME"] = shortHome.ToString();
+                eb["PATH"] = Interpreter.Info.BinPath + ";" + Environment.GetEnvironmentVariable("PATH");
 
-                psi.EnvironmentVariables["HOMEPATH"] = profilePath.Substring(2);
-                _sessionLogger.LogTrace(Resources.Trace_EnvironmentVariable, "HOMEPATH", psi.EnvironmentVariables["HOMEPATH"]);
+                _sessionLogger.LogInformation(Resources.Info_StartingRHost, Id, User.Name, rhostExePath, arguments);
+                Stream stderror;
+                int pid  = Win32Process.StartProcessAsUser(useridentity, rhostExePath, arguments, Path.GetDirectoryName(rhostExePath), eb, out stdin, out stdout, out stderror);
 
-                psi.EnvironmentVariables["USERPROFILE"] = $"{psi.EnvironmentVariables["HOMEDRIVE"]}{psi.EnvironmentVariables["HOMEPATH"]}";
-                _sessionLogger.LogTrace(Resources.Trace_EnvironmentVariable, "USERPROFILE", psi.EnvironmentVariables["USERPROFILE"]);
+                _process = Process.GetProcessById(pid);
+                _process.EnableRaisingEvents = true;
 
-                psi.EnvironmentVariables["APPDATA"] = $"{psi.EnvironmentVariables["USERPROFILE"]}\\AppData\\Roaming";
-                _sessionLogger.LogTrace(Resources.Trace_EnvironmentVariable, "APPDATA", psi.EnvironmentVariables["APPDATA"]);
+                _process.Exited += delegate {
+                    _hostEnd?.Dispose();
+                    _hostEnd = null;
+                    State = SessionState.Terminated;
+                };
 
-                psi.EnvironmentVariables["LOCALAPPDATA"] = $"{psi.EnvironmentVariables["USERPROFILE"]}\\AppData\\Local";
-                _sessionLogger.LogTrace(Resources.Trace_EnvironmentVariable, "LOCALAPPDATA", psi.EnvironmentVariables["LOCALAPPDATA"]);
+                _process.WaitForExit(250);
+                if (_process.HasExited && _process.ExitCode < 0) {
+                    var message = ErrorCodeConverter.MessageFromErrorCode(_process.ExitCode);
+                    if (!string.IsNullOrEmpty(message)) {
+                        throw new Win32Exception(message);
+                    }
+                    throw new Win32Exception(_process.ExitCode);
+                }
 
-                psi.EnvironmentVariables["TEMP"] = $"{psi.EnvironmentVariables["LOCALAPPDATA"]}\\Temp";
-                _sessionLogger.LogTrace(Resources.Trace_EnvironmentVariable, "TEMP", psi.EnvironmentVariables["TEMP"]);
+                _sessionLogger.LogInformation(Resources.Info_StartedRHost, Id, User.Name);
+
+                HostToClientErrorWorker(stderror, _process.Id, (int processid, string errdata) => {
+                    outputLogger?.LogTrace(Resources.Trace_ErrorDataReceived, processid, errdata);
+                }).DoNotWait();
+            } else {
```

Comment:
Can we unify both scenarios under the first branch? i.e. just use regular process creation Win32 function, but leave everything else the same? I'm worried that there will be subtly different behavior here due to differing implementations of pipes etc - the more can be unified, the less likely that is.